### ISSUE TITLE
Update tasks always fail

### ISSUE THREAD

**Issue by mglaman** [geerlingguy/ansible-role-drupal#36]
This is for an issue I found when working with DrupalVM (geerlingguy/drupal-vm#1284). Originally I had this issue when using DrupalVM as a multisite. However, this is happening for a single site install.

```
TASK [geerlingguy.drupal : include] ********************************************
included: /home/ubuntu/dashboard/vendor/geerlingguy/drupal-vm/provisioning/roles/geerlingguy.drupal/tasks/update.yml for xxx.xxx.xxx.xxx

TASK [geerlingguy.drupal : Run database updates.] ******************************
fatal: [xxx.xxx.xxx.xxx]: FAILED! => {"changed": true, "cmd": ["drush", "updatedb", "-y"], "delta": "0:00:00.286104", "end": "2017-04-18 16:42:11.340819", "failed": true, "rc": 1, "start": "2017-04-18 16:42:11.054715", "stderr": "Command updatedb needs a higher bootstrap level to run - you will        [error]\nneed to invoke drush from a more functional Drupal environment to run\nthis command.\nThe drush command 'updatedb' could not be executed.                      [error]", "stderr_lines": ["Command updatedb needs a higher bootstrap level to run - you will        [error]", "need to invoke drush from a more functional Drupal environment to run", "this command.", "The drush command 'updatedb' could not be executed.                      [error]"], "stdout": "", "stdout_lines": []}
```

I wonder if we should be using `--root=` versus changing the directory?

**Comment by mglaman** [comment]
Testing locally (because it happens locally and in CircleCI) with this modified `update.yml`

```
---
- name: Run database updates.
  command: drush updatedb -y --root={{ drupal_core_path }}

# TODO: Import configuration if configured?
# TODO: Other commands if configured?

- name: Rebuild Drupal caches.
  command: drush cache-rebuild --quiet --root={{ drupal_core_path }}
```

**Comment by geerlingguy** [comment]
```
Command updatedb needs a higher bootstrap level to run
```

Annoyingly, this is Drush's way of saying "for one of a thousand reasons, I couldn't actually connect to the Drupal database".

A lot of times, I notice this is an issue if you have (in `settings.php`) the database URL set to `localhost` instead of `127.0.0.1` or vice-versa. But it could be a number of other things.

If you run `drush updatedb -y` within `drupal_core_path` by hand, does it work?

**Comment by mglaman** [comment]
It works by hand, and I can confirm my suggestion fixed it.

```
TASK [geerlingguy.drupal : Define drupal_deploy_updated] ***********************
ok: [xxx.xxx.xxx.xxx]

TASK [geerlingguy.drupal : include] ********************************************
included: /Users/mglaman/Clients/commerceguys/reports-leancom/vendor/geerlingguy/drupal-vm/provisioning/roles/geerlingguy.drupal/tasks/update.yml for xxx.xxx.xxx.xxx

TASK [geerlingguy.drupal : Run database updates.] ******************************
changed: [xxx.xxx.xxx.xxx]

TASK [geerlingguy.drupal : Rebuild Drupal caches.] *****************************
changed: [xxx.xxx.xxx.xxx]
```

**Comment by mglaman** [comment]
I made a PR quick, because I have to run. I'm trying to think how I can get my project which has drupal-vm to use this PR as a patch, so I can test on CircleCI

**Comment by geerlingguy** [comment]
@mglaman - I wonder if maybe the project you're using is setting Drush up incorrectly or something like that... in drupalvm-live (https://github.com/geerlingguy/drupalvm-live) the tasks work correctly, as do they in Drupal VM's tests and this role's own tests.

Can you confirm that if you run the exact command `drush updb -y` inside the drupal core path directory (without the `--root`) argument, that _that_ actually works? I'm wondering if there's some environment issue, or something in your project's drush configuration (drushrc, aliases, or something else) that would be throwing drush for a loop.

Also, does the project have its own Drush copy installed (e.g. is it a Composer-based project and drush is a requirement?). If so, then this role might be picking up the project's Drush installation instead of the global one installed by Drupal VM itself.

Basically, I'd rather not add the `--root` argument if it can be avoided, because I have often had more trouble when explicitly setting `--root` :/

**Comment by mglaman** [comment]
I'll compare against drupalvm-live. It might be something I configured between my two setups (second basically a copy and paste job.)

I'm able to do `drush cim` etc from inside of my `drupal_core_path` path. 

Andddd I just realized it does have its own copy of Drush as a dependency. But that shouldn't run unless it explicitly did `./bin/drush`, as far as I know.

I'll try to run Ansible with some verbosity and force the update task to run. Because whatever I get figured out, I'd like to document in https://github.com/mglaman/commerce-project-template so there's another example in the wild.

I have to focus on some client work, but I'll circle back to this. If not now, then in Baltimore

**Comment by geerlingguy** [comment]
Sounds good! Even if the end result is switching to `--root` I want to make sure we have a good reason for it and figure out why the other way didn't work :)

**Comment by joestewart** [comment]
My understanding is that drush does try to use the project local install and not just when called directly.

This is part of the problem when using multisite without a --uri argument - https://github.com/drush-ops/drush/issues/2285

**Comment by mglaman** [comment]
I removed `drush/drush` from the project. Same results on CircleCI.

```
{"changed": true, "cmd": ["drush", "updatedb", "-y"], "delta": "0:00:00.254713", "end": "2017-04-18 22:12:48.687143", "failed": true, "rc": 1, "start": "2017-04-18 22:12:48.432430", "stderr": "Command updatedb needs a higher bootstrap level to run - you will        [error]\nneed to invoke drush from a more functional Drupal environment to run\nthis command.\nThe drush command 'updatedb' could not be executed.                      [error]", "stderr_lines": ["Command updatedb needs a higher bootstrap level to run - you will        [error]", "need to invoke drush from a more functional Drupal environment to run", "this command.", "The drush command 'updatedb' could not be executed.                      [error]"], "stdout": "", "stdout_lines": []}
```

I'll try to run it locally with more verbosity. Like having Drush run in verbose mode.

**Comment by oxyc** [comment]
@mglaman do you have the current code somewhere and I can try and reproduce it.

**Comment by mglaman** [comment]
Currently it's not up in public. It's some private code for Commerce Guys
demos at DrupalCom for Drupal Commerce 2. I'll work to put a public version
up.

**Comment by mglaman** [comment]
Here's a link to a gist of the configs: https://gist.github.com/mglaman/1f980989884ef79e6a41087c3b4731b5

**Comment by mglaman** [comment]
Also, FWIW ` "{{ drush_path }} -r {{ drupal_core_path }} core-cron"` is working fine

**Comment by mglaman** [comment]
Well this is a bummer. I added the following lines of debugging

```
- name: Debugging
  command: pwd
  args:
    chdir: "{{ drupal_core_path }}"
  register: pwd_output

- name: Show pwd output
  debug: msg="{{ pwd_output.stdout_lines }}"

- name: Debugging ls
  command: ls -l
  args:
    chdir: "{{ drupal_core_path }}"
  register: ls_output

- name: Show ls output
  debug: msg="{{ ls_output.stdout_lines }}"
```

And it looks right

```
ok: [104.236.244.212] => {
    "msg": [
        "/var/www/drupal/web"
    ]
}

ok: [104.236.244.212] => {
    "msg": [
        "total 44", 
        "-rw-rw-r--  1 drupalvm drupalvm  385 Apr 14 21:07 autoload.php", 
        "drwxrwxr-x 12 drupalvm drupalvm 4096 Apr 14 21:08 core", 
        "-rw-rw-r--  1 drupalvm drupalvm  549 Apr 14 21:07 index.php", 
        "drwxrwxr-x  4 drupalvm drupalvm 4096 Apr 14 21:08 modules", 
        "drwxrwxr-x  3 drupalvm drupalvm 4096 Apr 14 21:10 profiles", 
        "-rw-rw-r--  1 drupalvm drupalvm 1596 Apr 14 21:07 robots.txt", 
        "drwxrwxr-x  3 drupalvm drupalvm 4096 Apr 14 21:07 sites", 
        "drwxrwxr-x  4 drupalvm drupalvm 4096 Apr 14 21:08 themes", 
        "-rw-rw-r--  1 drupalvm drupalvm  848 Apr 14 21:07 update.php", 
        "-rw-rw-r--  1 drupalvm drupalvm 4555 Apr 14 21:07 web.config"
    ]
}
```

Still fails. I'll keep poking.

**Comment by joestewart** [comment]
@mglaman you may be familiar with this issue but it seems similar - https://github.com/drush-ops/drush/issues/2067

**Comment by geerlingguy** [comment]
@mglaman - I'll tag a release shortly (one final bit of cleanup first)—did you test the change and confirm it runs drush properly?

**Comment by mglaman** [comment]
@geerlingguy As far as I can tell, yes.

**Comment by geerlingguy** [comment]
Sweet, thanks for confirming, closing the issue and will merge into Drupal VM in a few minutes!
